src/NuGet.Clients/NuGet.VisualStudio.Internal.Contracts/INuGetProjectManagerService.cs

```diff
@@ -65,5 +65,19 @@ public interface INuGetProjectManagerService : IDisposable
             DependencyBehavior dependencyBehavior,
             IReadOnlyList<string> packageSourceNames,
             CancellationToken cancellationToken);
+
+        /// <summary>
+        /// Gets a list of packages that indirectly installs a given transitive package in a project
+        /// </summary>
+        /// <param name="transitivePackage">Transitive Package coordinates (packageId, Version)</param>
+        /// <param name="projectId">Internal Project ID to retreive packages from</param>
+        /// <param name="cancellationToken">Cancelation token</param>
+        /// <returns>A Dictionary of user-installed packages that depends on the transitive package, indexed by framework/runtime combination</returns>
+        /// <exception cref="OperationCanceledException">If cancellation token is signaled to cancel</exception>
+        /// <remarks>Only work for PackageReference-based projects. Otherwise, it will return an empty dictionary</remarks>
+        ValueTask<IReadOnlyDictionary<Tuple<NuGetFramework, string>, IReadOnlyList<IPackageReferenceContextInfo>>> GetTransitivePackageOriginAsync(
```

Comment:
The context in which this API will be used is to populate the installed tab.
In every scenario that I can think of, we always want, all transitive packages and their respective top level packages. 

Would it be a better idea to add an API that returns *all* transitive packages and their respective top level packages?

Reply:
You can achieve that by calling `GetInstalledAndTransitivePackagesAsync()` and this new API.

I am under the impression that this API will be callen when the user points the mouse to a transitive dependency in the PM UI to show its top level packages in a tooltip.

So, do you want to precompute all top level packages given all transitive packages?

Reply:
hmm, that's a good point. 

I think as you work on the caching later on, it'll be better to precompute things, but all of that can be done in the CPS/Legacy packageReference projects. 

I think you're right though. 

Consider this not relevant. I'll take another look.